tensorpipe/CMakeLists.txt

```diff
@@ -120,6 +121,11 @@ target_link_libraries(tensorpipe PRIVATE protobuf::libprotobuf)
 protobuf_generate(TARGET tensorpipe)
 
 
+if(BUILD_SHARED_LIBS)
+  set_target_properties(tensorpipe PROPERTIES POSITION_INDEPENDENT_CODE 1)
+endif()
```

Comment:
Also from the CMake docs:
> For SHARED and MODULE libraries the POSITION_INDEPENDENT_CODE target property is set to ON automatically.

What we might want to enable though is for the user to say "I want a static lib, but with PIC".

Reply:
Yes, this is the gist. Note that this is in fact exactly what we already do in our own target for libuv (i.e., `_uv_a`), so that when TensorPipe is built dynamically we still build libuv statically but with PIC so that it can be linked against the dynamic libtensorpipe.

Reply:
But here tensorpipe will only get build with PIC if `BUILD_SHARED_LIBS` is `ON`?

Reply:
Correct: if the global BUILD_SHARED_LIBS, which is possibly coming from a parent project, is ON, then TensorPipe will be built with PIC, even when TensorPipe is a static library due to TP_STATIC_OR_SHARED being set to STATIC by the parent project.